api/lib/domain/services/certification-candidates-ods-service.js

```diff
@@ -32,23 +38,40 @@ async function extractCertificationCandidatesFromCandidatesImportSheet({ session
 
   certificationCandidatesDataByLine = _filterOutEmptyCandidateData(certificationCandidatesDataByLine);
 
-  return _.map(certificationCandidatesDataByLine, (certificationCandidateData, line) => {
+  return await Promise.all(_.map(certificationCandidatesDataByLine, async (certificationCandidateData, line) => {
```

Comment:
Surtout pas de Promise.all quand la callback fait des appels BDD ! Risque de saturation du pool de connexions ! Il faut favoriser le bluebird.mapSeries, les faire les uns à la suite des autres est aussi bien.
On pourrait presque envisager de lire les tables countries et cities en amont une seule fois puis les exploiter pour chaque ajout candidat ? (voire mettre dans Redis ?? v1 of course)

Reply:
Ok j'y avais pas pensé, je vais refacto avec un bluebird.mapSeries alors.